silo/views.py

```diff
@@ -850,11 +850,17 @@ def uploadFile(request, id):
             silo.reads.add(read_obj)
             silo_id = silo.id
 
+            task = CeleryTask(task_id=None,
+                              task_status=None,
+                              content_object=read_obj)
+            task.save()
+
             async_res = process_silo.apply_async(
                         (silo.id, read_obj.id)
             )
 
-            task = CeleryTask(task_id=async_res.id, task_status=CeleryTask.TASK_CREATED, content_object=read_obj)
+            task.task_id = async_res.id
+            task.task_status = CeleryTask.TASK_CREATED
```

Comment:
Does it solve the problem? It looks like the same thing for me.
- Before you were creating the task with values for the fields
- Now, you're creating the task and updating the fields afterward.

Another things: You should use `CeleryTask.objects.create()` instead of `save()` here.

Reply:
I am not 100% sure since i cannot reproduce the error locally. But the issue was that the CeleryTask (our own db model representing the task) did not exist when the actual task was being executed.
Before the changes it was "Create Task in Queue" -> "Save Task id in new CeleryTask to db" but when executing the queue the CeleryTask somehow does not exist yet. but it exists when i check in the db later, so it is getting saved, but not fast enough.
So i try to solve the problem by first creating the object and saving it to the database empty and then sending the task to the queue.
Later i update the task in the db with the task_id which is not critical to have at that point. So the process is now "Create empty `CeleryTask`" -> "Create Task in Queue" -> "Update CeleryTask to have `task_id`"